scripts/e2e.sh

```diff
@@ -0,0 +1,374 @@
+#!/usr/bin/env bash
+set -xeuf
+
+# This is the full workflow test of the ilab CLI.
+#
+# We expect it to be run anywhere `ilab` would run, including the instructlab
+# container images.
+#
+# It represents the workflow a typical user would run through with ilab.
+
+HF_TOKEN=${HF_TOKEN:-}
+GRANITE_7B_MODEL="instructlab/granite-7b-lab"
+MIXTRAL_8X7B_MODEL="mistralai/Mixtral-8x7B-Instruct-v0.1"
+PROMETHEUS_8X7B_MODEL="prometheus-eval/prometheus-8x7b-v2.0"
+TRAINED_MODEL_PATH=""
+LARGE=0
+
+BOLD='\033[1m'
+NC='\033[0m' # No Color
+
+SCRIPTDIR=$(dirname "$0")
+E2E_TEST_DIR=""
+CONFIG_HOME=""
+DATA_HOME=""
+CACHE_HOME=""
+CONFIG_HOME=""
+
+check_flags() {
+    if [ "${LARGE}" -ne 1 ]; then
+         echo "ERROR: Must specify a flag when invoking this script."
+         usage
+         exit 1
+    fi
+
+}
+
+########################################
+# This function initializes the directory
+# used during end-to-end testing. This function
+# also creates the directory structure which
+# instructlab expects to see at the time of execution.
+# Arguments:
+#   None
+# Globals:
+#   DATA_HOME
+#   CACHE_HOME
+#   CONFIG_HOME
+#   E2E_TEST_DIR
+#   HOME
+# Outputs:
+#   None
+########################################
+
+init_e2e_tests() {
+    E2E_TEST_DIR=$(mktemp -d)
+    HOME="${E2E_TEST_DIR}"  # update the HOME directory used to resolve paths
+
+    CONFIG_HOME=$(python -c 'import platformdirs; print(platformdirs.user_config_dir())')
+    DATA_HOME=$(python -c 'import platformdirs; print(platformdirs.user_data_dir())')
+    CACHE_HOME=$(python -c 'import platformdirs; print(platformdirs.user_cache_dir())')
+    # ensure that our mock e2e dirs exist
+    for dir in "${CONFIG_HOME}" "${DATA_HOME}" "${CACHE_HOME}"; do
+    	mkdir -p "${dir}"
+    done
+
+    E2E_PATH_DIR="${HOME}/bin"
+    E2E_LOG_DIR="${HOME}/log"
+    mkdir -p "${E2E_PATH_DIR}"
+    mkdir -p "${E2E_LOG_DIR}"
+    PATH=${E2E_PATH_DIR}:$PATH
+    wget https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64 -O "${E2E_PATH_DIR}/yq" && chmod +x "${E2E_PATH_DIR}/yq"
```

Comment:
There is a risk here with running untrusted binaries. Could you use the python version here (pip install yq) or alternatively use go install if you want to use this version (you probably have to dnf install golang first)

Reply:
I'll take a look at using one of the language specific package managers.

I don't feel great about downloading and using yq in that function but ideally most if not all of the config overwriting logic in that function would go away once we have system profiles.